File: showly_flutter/lib/features/shows/data/models/show_model.dart
import '../../../../core/utils/clean_html.dart';
import '../../domain/entities/show.dart';

class ShowModel {
  final Map<String, dynamic> json;

  ShowModel(this.json);

  Show toEntity() {
    final image = json['image'] as Map<String, dynamic>?;

    return Show(
      id: json['id'] ?? DateTime.now().millisecondsSinceEpoch,
      name: json['name'] ?? 'Untitled show',
      imageUrl:
          image?['original'] ??
          image?['medium'] ??
          'https://static.tvmaze.com/images/no-img/no-img-portrait-text.png',
      summary: cleanHtml(json['summary']),
      genres: List<String>.from(json['genres'] ?? []),
      rating: (json['rating']?['average'] as num?)?.toDouble(),
      status: json['status'] ?? 'Unknown',
      language: json['language'] ?? 'Unknown',
      runtime: json['runtime'],
      premiered: json['premiered'],
      isLocal: false,
    );
  }
}

File: showly-react-native/src/features/shows/data/models/showModel.ts
import { Show } from "../../domain/entities/show";
import { cleanHtml } from "../../../../core/utils/cleanHtml";

export type TvMazeShowModel = {
  id: number;
  name: string;
  image?: {
    medium?: string;
    original?: string;
  } | null;
  summary?: string | null;
  genres?: string[];
  rating?: {
    average?: number | null;
  };
  status?: string;
  language?: string;
  runtime?: number | null;
  premiered?: string | null;
};

export function mapTvMazeShowToEntity(model: TvMazeShowModel): Show {
  return {
    id: model.id,
    name: model.name,
    imageUrl:
      model.image?.original ??
      model.image?.medium ??
      "https://static.tvmaze.com/images/no-img/no-img-portrait-text.png",
    summary: cleanHtml(model.summary),
    genres: model.genres ?? [],
    rating: model.rating?.average ?? null,
    status: model.status ?? "Unknown",
    language: model.language ?? "Unknown",
    runtime: model.runtime ?? null,
    premiered: model.premiered ?? null,
    isLocal: false,
  };
}